File: memories/server/models/postMessage.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

// Post represents the structure of a post document in MongoDB
type Post struct {
	ID           primitive.ObjectID `json:"id,omitempty" bson:"_id,omitempty"`
	Title        string             `json:"title" bson:"title"`
	Message      string             `json:"message" bson:"message"`
	Creator      string             `json:"creator" bson:"creator"`
	Tags         []string           `json:"tags" bson:"tags"`
	SelectedFile string             `json:"selectedFile" bson:"selectedFile"`
	LikeCount    int                `json:"likeCount" bson:"likeCount"`
	CreatedAt    time.Time          `json:"createdAt" bson:"createdAt"`
}

File: memories/server/models/postMessage.ts
import mongoose, { Document, Schema, Model } from 'mongoose';

// Define an interface that describes the structure of a post document
interface IPost extends Document {
    title: string;
    message: string;
    creator: string;
    tags: string[];
    selectedFile: string;
    likeCount: number;
    createdAt: Date;
}

// Define the schema with type annotations
const postSchema: Schema<IPost> = new mongoose.Schema({
    title: { type: String, required: true },
    message: { type: String, required: true },
    creator: { type: String, required: true },
    tags: { type: [String], required: true },
    selectedFile: { type: String },
    likeCount: {
        type: Number,
        default: 0,
    },
    createdAt: {
        type: Date,
        default: new Date(),
    },
});

// Create the Post model
const PostMessage: Model<IPost> = mongoose.model<IPost>('PostMessage', postSchema);

export default PostMessage;
